src/components/execution-environment/repository-form.tsx

```diff
@@ -343,12 +353,34 @@ export class RepositoryForm extends React.Component<IProps, IState> {
                 )
               }
             ></ObjectPermissionField>
+            {formError.length > 0 &&
+              formError.map((error) => (
+                <Alert title={error.title} variant='danger' isInline>
+                  {error.detail}
+                </Alert>
+              ))}
           </FormGroup>
         </Form>
       </Modal>
     );
   }
 
+  private validateName(name) {
+    const regex = /^([0-9A-Za-z._-]+\/)?[0-9A-Za-z._-]+$/;
+    if (regex.test(name)) {
+      return 'default';
+    } else {
+      return 'error';
+    }
+  }
+
+  private formIsValid() {
+    return (
+      !this.props.isNew ||
+      (this.validateName(this.state.name) && !!this.state.name)
```

Comment:
should this be..

```suggestion
      (this.state.name && this.validateName(this.state.name))
```

to prevent running validation on empty names?
Or it could be just

```suggestion
      this.validateName(this.state.name)
```

since the regex will return false on null and empty string?